protocol/messenger_contact_requests_test.go

```diff
@@ -1276,3 +1276,398 @@ func (s *MessengerContactRequestSuite) TestPairedDevicesRemoveContact() {
+	s.Require().Len(resp.Contacts, 1)
+
+	// Check the contact state is correctly set
+	s.Require().Equal(ContactRequestStateNone, resp.Contacts[0].ContactRequestLocalState)
+	s.Require().Equal(ContactRequestStateReceived, resp.Contacts[0].ContactRequestRemoteState)
+}
+
+// The scenario tested is as follow:
+// 1) Alice sends a contact request to Bob
+// 2) Bob accepts the contact request
+// 3) Bob goes offline
+// 4) Alice retracts the contact request
+// 5) Alice adds bob back to her contacts
+// 6) Bob goes online, they receive 4 and 5 in the correct order
+func (s *MessengerContactRequestSuite) TestAliceOfflineRetractsAndAddsCorrectOrder() {
+	// Alice sends a contact request to bob
+	alice1 := s.m
+
+	bob := s.newMessenger(s.shh)
+	_, err := bob.Start()
+	s.Require().NoError(err)
+
+	bobID := types.EncodeHex(crypto.FromECDSAPub(&bob.identity.PublicKey))
+	myID := types.EncodeHex(crypto.FromECDSAPub(&alice1.identity.PublicKey))
+
+	request := &requests.AddContact{
+		ID: types.Hex2Bytes(bobID),
+	}
+
+	_, err = alice1.AddContact(context.Background(), request)
+	s.Require().NoError(err)
+
+	// Wait for the message to reach its destination
+	resp, err := WaitOnMessengerResponse(
+		bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Contacts) > 0 && len(r.Messages()) > 0 && len(r.ActivityCenterNotifications()) > 0
+		},
+		"no messages",
+	)
+
+	// Check contact request has been received
+	s.Require().NoError(err)
+
+	// Check the contact state is correctly set
+	s.Require().Len(resp.Contacts, 1)
+	s.Require().Equal(ContactRequestStateReceived, resp.Contacts[0].ContactRequestRemoteState)
+
+	// Bob accepts the contact request
+	_, err = bob.AcceptLatestContactRequestForContact(context.Background(), &requests.AcceptLatestContactRequestForContact{ID: types.Hex2Bytes(myID)})
+	s.Require().NoError(err)
+
+	// Alice receives the accepted confirmation
+	resp, err = WaitOnMessengerResponse(
+		alice1,
+		func(r *MessengerResponse) bool {
+			return len(r.Contacts) > 0
+		},
+		"no messages",
+	)
+	s.Require().NoError(err)
+
+	// Make sure we consider them a mutual contact, sender side
+	mutualContacts := alice1.MutualContacts()
+	s.Require().Len(mutualContacts, 1)
+
+	// Check the contact state is correctly set
+	s.Require().Len(resp.Contacts, 1)
+	s.Require().True(resp.Contacts[0].mutual())
+
+	_, err = alice1.RetractContactRequest(&requests.RetractContactRequest{ContactID: types.Hex2Bytes(bob.myHexIdentity())})
+	s.Require().NoError(err)
+
+	// adds bob again to her device
+	request = &requests.AddContact{
+		ID: types.Hex2Bytes(bobID),
+	}
+
+	_, err = alice1.AddContact(context.Background(), request)
+	s.Require().NoError(err)
+
+	// Wait for the message to reach its destination
+	_, err = WaitOnMessengerResponse(
+		bob,
+		func(r *MessengerResponse) bool {
+			return len(r.ActivityCenterNotifications()) > 0
+		},
+		"no messages",
+	)
+	s.Require().NoError(err)
+
```

Comment:
Did you forget here to test Bob's side?

Reply:
No, we are happy with just checking that they receive a notificaton, which is tested otherwise `err` would be nil